core/src/main/java/org/apache/iceberg/FilteredManifest.java

```diff
@@ -95,60 +95,31 @@ public FilteredManifest caseSensitive(boolean isCaseSensitive) {
       Evaluator evaluator = evaluator();
       InclusiveMetricsEvaluator metricsEvaluator = metricsEvaluator();
 
-      return CloseableIterable.filter(reader.entries(projection(fileSchema, columns, caseSensitive)),
+      // ensure stats columns are present for metrics evaluation
+      Collection<String> projectColumns = withStatsColumns(columns);
```

Comment:
This is good, but can we make sure there is a row filter before adding these columns? Without the row filter they aren't needed.

Reply:
Fixed